store/sqlstore/channel_store.go

```diff
@@ -2850,47 +2850,93 @@ func (s SqlChannelStore) AutocompleteInTeam(teamID, userID, term string, include
-		WHERE
-			(C.TeamId = :TeamId OR (C.TeamId = '' AND C.Type = :ChannelType))
-			AND CM.UserId = :UserId
-			` + deleteFilter + `
-			%v
-		LIMIT 50`
+	// shared query
+	baseQuery := s.getSubQueryBuilder().Select("C.*").
+		From("Channels AS C").
+		Join("ChannelMembers AS CM ON CM.ChannelId = C.Id").
+		Limit(50)
 
-	var channels model.ChannelList
+	var (
+		channels model.ChannelList
+		sql      string
+		args     []interface{}
+	)
 
-	if likeClause, likeTerm := s.buildLIKEClause(term, "Name, DisplayName, Purpose"); likeClause == "" {
-		if _, err := s.GetReplica().Select(&channels, fmt.Sprintf(queryFormat, ""), map[string]interface{}{"TeamId": teamId, "UserId": userId, "ChannelType": model.ChannelTypeGroup}); err != nil {
-			return nil, errors.Wrapf(err, "failed to find Channels with term='%s'", term)
+	// build the like clause
+	like := s.buildLIKEClauseX(term, "Name", "DisplayName", "Purpose")
+	if like == nil {
+		var err error
+
+		// use the base query for this request
+		query := baseQuery.Where(baseWhere)
+
+		// generate the SQL query
+		sql, args, err = query.ToSql()
+		if err != nil {
+			return nil, errors.Wrap(err, "AutocompleteInTeamForSearch_Tosql")
 		}
 	} else {
+		// build the full text search clause
+		full := s.buildFulltextClauseX(term, "Name", "DisplayName", "Purpose")
+		if full == nil {
+			return nil, errors.New("failed to build full text query for term=" + term)
+		}
+
+		// build the LIKE query
+		likeWhere := append(baseWhere, like)
+		likeSQL, likeArgs, err := baseQuery.Where(likeWhere).ToSql()
+		if err != nil {
+			return nil, errors.Wrap(err, "AutocompleteInTeamForSearch_Like_Tosql")
+		}
+
+		// build the full text query
+		fullWhere := append(baseWhere, full)
+		fullSQL, fullArgs, err := baseQuery.Where(fullWhere).ToSql()
+		if err != nil {
+			return nil, errors.Wrap(err, "AutocompleteInTeamForSearch_Full_Tosql")
+		}
+
 		// Using a UNION results in index_merge and fulltext queries and is much faster than the ref
 		// query you would get using an OR of the LIKE and full-text clauses.
-		fulltextClause, fulltextTerm := s.buildFulltextClause(term, "Name, DisplayName, Purpose")
-		likeQuery := fmt.Sprintf(queryFormat, "AND "+likeClause)
-		fulltextQuery := fmt.Sprintf(queryFormat, "AND "+fulltextClause)
-		query := fmt.Sprintf("(%v) UNION (%v) LIMIT 50", likeQuery, fulltextQuery)
+		sql = fmt.Sprintf("(%s) UNION (%s) LIMIT 50", likeSQL, fullSQL)
+		args = append(likeArgs, fullArgs...)
+	}
+
+	var err error
 
-		if _, err := s.GetReplica().Select(&channels, query, map[string]interface{}{"TeamId": teamId, "UserId": userId, "LikeTerm": likeTerm, "FulltextTerm": fulltextTerm, "ChannelType": model.ChannelTypeGroup}); err != nil {
-			return nil, errors.Wrapf(err, "failed to find Channels with term='%s'", term)
+	// since the UNION is not part of squirrel, we need to assemble it and then update
+	// the placeholders manually
+	if s.DriverName() == model.DatabaseDriverPostgres {
```

Comment:
Sorry, I don't understand what the issue is here.

Reply:
Aha! I missed that you are using `subQueryBuilder` instead of `queryBuilder`. `subQueryBuilder` will always generate in Question format but `queryBuilder` will generate in question or dollar format depending on the DB. If we used `queryBuilder`, it wouldn't have worked because replacePlaceholders will only work on question format.

Reply:
Yes, I ran into that problem when testing. Then I discovered the `subQueryBuilder` and it works like a charm.